adapters/src/main/java/org/jboss/jca/adapters/jdbc/BaseWrapperManagedConnection.java

```diff
@@ -622,9 +610,8 @@ private void returnHandle(WrappedConnection handle, boolean error)
 
          ce.setConnectionHandle(handle);
 
-         for (Iterator<ConnectionEventListener> i = copy.iterator(); i.hasNext();)
+         for (ConnectionEventListener cel : cels)
```

Comment:
I'm not sure about that change. It will render the iteration unsafe, i.e, subject to IndexOutOfBoundsException when doing cels.get(i)
I can keep the Iterator explicitly, but I don't see any reason to not use the enhanced for loop.